Api/V8/Service/ModuleService.php

```diff
@@ -111,23 +115,88 @@ public function getRecords(GetModulesParams $params, Request $request)
+        ) {
+            $selectedFields = strtolower($module) . '.' . implode(','
+                    . strtolower($module[0]) . '.', $fields);
+            $selectedModule = strtolower($module);
+
+            // Selects Module or COUNT(*) and will add one to the query.
+            $idSelect = 'SELECT ' . $selectedModule . '.id ';
+            $countSelect = 'SELECT COUNT(*) AS cnt ';
+
+            $quotedCountSelect = $db->quote($countSelect);
+
+            // Email where clause
+            $fromQuery
+                = 'FROM email_addresses join email_addr_bean_rel on email_addresses.id = email_addr_bean_rel.email_address_id join '
+                . $selectedModule . ' on ' . $selectedModule
+                . '.id = email_addr_bean_rel.bean_id ';
+            $modifiedWhere = str_replace('accounts.email1',
+                'email_addresses.email_address', $where);
+            $where = (string)$modifiedWhere;
+
+            /** @noinspection TypeUnsafeComparisonInspection */
+            // Sets and adds deleted to the query
+            if ($deleted == 0) {
+                $whereAuto = '' . $bean->table_name . ' .deleted=0';
+            } elseif ($deleted == 1) {
+                $whereAuto = '' . $bean->table_name . ' .deleted=1';
+            }
+            if ($where != '') {
+                $where = ' where (' . $where . ') AND ' . $whereAuto . '';
+            } else {
+                $where = ' where ' . $whereAuto . '';
+            }
+
+            // Joins parts together to form query
+            $query = $idSelect . $fromQuery . $where;
+            $countQuery = $quotedCountSelect . $fromQuery . $where;
+            $realRowCount = (int)$db->fetchRow($db->query($countQuery, true,
+                ''))['cnt'];
+
+            // Sets orderby into the query
+            $order_by = $bean->process_order_by($orderBy);
+            if (!empty($orderBy)) {
+                $query .= ' ORDER BY ' . $order_by;
+            }
+
+            $result = $bean->process_list_query($query, $offset, $limit, -1,
+                $where);
+            $beanResult['row_count'] = $result['row_count'];
+            $beanList = [];
 
+            foreach ($result['list'] as $resultBean) {
+                $queryModuleBean = \BeanFactory::newBean($module);
+                $queryModuleBean->id = $resultBean->id;
+                $beanList[] = $queryModuleBean;
+            }
+            $beanResult['list'] = $beanList;
+            $beanListResponse = new BeanListResponse($beanResult);
+        } else {
+            $beanListResponse = $this->beanManager->getList($module)
+                ->orderBy($orderBy)
+                ->where($where)
+                ->offset($offset)
+                ->limit($limit)
+                ->max($size)
+                ->deleted($deleted)
+                ->fields($this->beanManager->filterAcceptanceFields($bean,
+                    $fields))
+                ->fetch();
+        }
```

Comment:
This if/else statement is very long and has high complexity. Consider splitting it into several methods.

Reply:
from speaking to Dillon this function is on the line from a cyclomatic complexity stand point, so this should potentially be looked at in the future if anything more is added to it.